variables.tf

```diff
@@ -15,3 +17,87 @@ variable "env" {
   type        = string
   default     = "default"
 }
+
+###############################################
+# Networking
+###############################################
+variable "vpc_id" {
+  description = "VPC ID"
+  type        = string
+}
+
+variable "subnet_ids" {
+  description = "List of subnet IDs to make Elasticache available on"
+  type        = list(string)
+}
+
+variable "ingress_rules" {
+  description = "List of ports to allow ingress on"
+  type = list(object({
+    port        = number
+    protocol    = string
+    cidr_blocks = list(string)
+  }))
+  default = [{
+    port        = 6379
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }]
+}
+
+###############################################
```

Comment:
also, also, we need a script to do this

Reply:
Thank the internet https://github.com/hashicorp/terraform/issues/12959

Reply:
I took this from that issue thread
```
#!/usr/bin/env python
# sort terraform variables

from __future__ import print_function
import sys
import re

# this regex matches terraform variable definitions
# we capture the variable name so we can sort on it
pattern = r'(variable ")([^"]+)(" {[^{]+})'


def process(content):
    # sort the content (a list of tuples) on the second item of the tuple
    # (which is the variable name)
    matches = sorted(re.findall(pattern, content), key=lambda x: x[1])

    # iterate over the sorted list and output them
    for match in matches:
        print(''.join(map(str, match)))

        # don't print the newline on the last item
        if match != matches[-1]:
            print()


# check if we're reading from stdin
if not sys.stdin.isatty():
    stdin = sys.stdin.read()
    if stdin:
        process(stdin)

# process any filenames on the command line
for filename in sys.argv[1:]:
    with open(filename) as f:
        process(f.read())

```

Reply:
I'm saving it for later, I'd like to make a pre-commit hook out of it: https://gist.github.com/sblack4/34d74f6a4a6df65eb8d6e563a5135111

Reply:
Then we can also sort the `description`, `type`, and `default` attributes on commit!!!

Reply:
I don't think it'll work because of multi-line vars, blocks, etc. Worth a shot but I'm not holding my breath. I just create them in alphabetical order which is even better than a script :)